internal/mithril-persistence/src/database/version_checker.rs

```diff
@@ -153,6 +155,53 @@ insert into db_version (application_type, version, updated_at) values ('{applica
 
         Ok(())
     }
+
+    fn check_minimum_required_version(
+        &self,
+        db_version: DbVersion,
+        migration: &SqlMigration,
+    ) -> StdResult<()> {
+        if db_version == 0 {
+            return Ok(());
+        }
+
+        if let Some(min_requirement) = &migration.minimum_requirement {
+            let min_required_version = migration.version - 1;
+            if db_version < min_required_version {
+                return Err(anyhow!(
+                    r#"
+                        Minimum required database version is not met to apply migration '{}'.
+                        Please migrate your {} node database with the minimum node version compatible '{}' available in the release: '{}'.
+
+                        First, download the required node version in your current directory by running the following command:
+                        curl --proto '=https' --tlsv1.2 -sSf https://raw.githubusercontent.com/input-output-hk/mithril/refs/heads/main/mithril-install.sh | sh -s -- -c mithril-{} -d {} -p $(pwd)
+
+                        Then run the database migrate command:
+                        mithril-{} database migrate --stores-directory /path/to/stores-directory
+                    "#,
+                    migration.version,
+                    self.application_type.to_string(),
+                    min_requirement.node_version,
+                    min_requirement.release_version,
+                    self.application_type.to_string(),
+                    min_requirement.release_version,
+                    self.application_type.to_string()
+                ));
+            }
+        }
+
+        Ok(())
+    }
+}
+
+/// Node version requirement for a migration.
+#[derive(Debug, Clone)]
+pub struct NodeVersionRequirement {
+    /// The node version required to apply a migration.
+    pub node_version: Version,
+
+    /// The release version containing the required node version.
+    pub release_version: String,
```

Comment:
We use the term "distribution" instead of release in our process, furthermore we can't specify the node version here as it won't works with migrations that are shared among multiple nodes (ie: transactions database migrations).
```suggestion
    /// The distribution version the user can fallback to in order to update his database before updating to the latest node.
    pub fallback_distribution_version: String,
```